### PR TITLE
[OSD-12925] feat(support): Limited Support to allow parameters

### PR DESCRIPTION
**Pull request by Tessg22**
This change enables `osdctl cluster support` command to accept the parameterized input. Doc included.

### CODE REVIEW

File: cmd/cluster/support/post.go [hunk 1]
```diff
@@ -89,13 +92,21 @@ func (o *postOptions) run() error {
...
 	// Check that the cluster key (name, identifier or external identifier) given by the user
 	// is reasonably safe so that there is no risk of SQL injection
 	err := ctlutil.IsValidClusterKey(o.clusterID)
 	if err != nil {
 		return err
 	}
 
+	// For every '-p' flag, replace it's related placeholder in the template
+	for k := range userParameterNames {
+		replaceFlags(userParameterNames[k], userParameterValues[k])
```
Comment (typeid) [inline cmd/cluster/support/post.go:107]: To me, `replaceFlags` doesn't describe the function well enough, how about:

**suggestion:** Rename this function to `insertFlags`

File: docs/command/osdctl_cluster_support.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+## osdctl cluster support
```
Comment (typeid) [inline docs/command/osdctl_cluster_support.md:1]: **Question:** do you know how this is generated? I would suppose by `make`, but for some newer commands, this isn't auto generated.

**Comment by typeid** [comment]
All threads opened are non blocking.
/lgtm

File: cmd/cluster/support/post.go (hunk 1 above)
Comment (Tessg22) [reply cmd/cluster/support/post.go:107]: The reason behind the wording `replaceFlags` is that the template contains the variables such as `${CONFIGURATION}` and that is _replaced_ by the input given in the command. When you drill down, it uses `strings.ReplaceAll`.

File: docs/command/osdctl_cluster_support.md (hunk 2 above)
Comment (Tessg22) [reply docs/command/osdctl_cluster_support.md:1]: It is generated with `make build` as per [osdctl#installation](https://github.com/openshift/osdctl#installation).

File: cmd/cluster/support/post.go (hunk 1 above)
Comment (typeid) [reply cmd/cluster/support/post.go:107]: Would `replaceWithFlags` fit even better then? Semantically, `replaceFlags` would mean that we replace the flags, not that we replace the variables with flags (which is why I initially suggested `insertFlags`)

File: docs/command/osdctl_cluster_support.md (hunk 2 above)
Comment (typeid) [reply docs/command/osdctl_cluster_support.md:1]: Thank you, I suppose something is broken with the new commands then, as the doc is not generated for them. I will look into it later if I have time.

File: cmd/cluster/support/post.go (hunk 1 above)
Comment (Tessg22) [reply cmd/cluster/support/post.go:107]: `replaceWithFlags` sounds good.

File: docs/command/osdctl_cluster_support.md (hunk 2 above)
Comment (Tessg22) [reply docs/command/osdctl_cluster_support.md:1]: My bad, I didn't notice that this is a comment for a doc file change.

File: cmd/cluster/support/post.go [hunk 3]
```diff
@@ -23,9 +24,10 @@ import (
 )
 
 var (
```
[the remark this answers is not in the document]
Comment (Tessg22) [reply cmd/cluster/support/post.go:26]: The reason for defining the LimitedSupport variable isn't unit testing but saving a template file that has been provided through the command into a JSON struct.

[the remark this answers is not in the document]
Comment (typeid) [reply cmd/cluster/support/post.go:26]: It's indeed defined to parse the template file to a struct, but it wouldn't have to be a global variable if it wasn't used in the unit tests (`post_test.go`: e.g L68, L74).

If the above was not the case, we could have defined all the global variables:
```
var (LimitedSupport                                          support.LimitedSupport
	template                                                string
	isDryRun                                                bool
	templateParams, userParameterNames, userParameterValues []string
)
```
within the `postOptions` struct.

The benefits would be:
- all variables in one place
- the json marshalling could be done before running any other logic (preemptive validation)
- no polluting of the global `support` namespace

This is out of scope for this PR though, and would require reworking the way we unit tests this. So just a thought :)

[the remark this answers is not in the document]
Comment (Tessg22) [reply cmd/cluster/support/post.go:26]: I understand now. Do you want to create an OSD card for it that can be picked up during the maintenance sprint?

**Comment by Tessg22** [comment]
@typeid could you please add also approve label?

**Comment by clcollins** [comment]
It looks like the SOP for Limited Support beat this change into osdctl.  We should be more careful to hold those PRs to OPS-SOP until the code it references is in place.

I was one of the approvers, so this is on me too, but we need to share this warning the rest of the team as well.

**Pull request merged**